ravenpy/utilities/geo.py

```diff
@@ -0,0 +1,238 @@
+import json
+import logging
+from pathlib import Path
+from typing import Union, List
+
+try:
+    import fiona
+    import rasterio
+    import rasterio.mask
+    import rasterio.vrt
+    import rasterio.warp
+    from pyproj import CRS
+    from shapely.geometry import GeometryCollection, shape, Polygon, MultiPolygon, mapping
+    from shapely.ops import transform
+except (ImportError, ModuleNotFoundError) as e:
+    msg = (
+        f"`{Path(__file__).stem}` requires installation of the RavenPy GIS libraries. These can be installed using the"
+        " `pip install ravenpy[gis]` recipe or via Anaconda (`conda env -n ravenpy-env -f environment.yml`)"
+        " from the RavenPy repository source files."
+    )
+    raise ImportError(msg) from e
+
+RASTERIO_TIFF_COMPRESSION = "lzw"
+LOGGER = logging.getLogger("RavenPy")
+WGS84 = 4326
+
+
+def geom_transform(
+    geom: Union[GeometryCollection, shape],
+    source_crs: Union[str, int, CRS] = WGS84,
+    target_crs: Union[str, int, CRS] = None,
+) -> GeometryCollection:
+    """Change the projection of a geometry.
+
+    Assuming a geometry's coordinates are in a `source_crs`, compute the new coordinates under the `target_crs`.
+
+    Parameters
+    ----------
+    geom : Union[GeometryCollection, shape]
+      Source geometry.
+    source_crs : Union[str, int, CRS]
+      Projection identifier (proj4) for the source geometry, e.g. '+proj=longlat +datum=WGS84 +no_defs'.
+    target_crs : Union[str, int, CRS]
+      Projection identifier (proj4) for the target geometry.
+
+    Returns
+    -------
+    GeometryCollection
+      Reprojected geometry.
+    """
+    try:
+        from pyproj import Transformer  # noqa
+        from functools import partial
+
+        source = (
+            CRS.from_epsg(source_crs)
+            if isinstance(source_crs, int or str)
+            else source_crs
```

Comment:
Personally I find that using single line if/else statements like that is less readable (just my opinion though!)

Reply:
I can imagine @huard also saying this. Will adjust.

Reply:
ok just to clarify because I realize what I said: I absolutely love conditional expressions in Python! I just find that when they span multiple lines they become more confusing though